packages/extension-chakra-store-locator/src/components/with-optional-commerce-sdk-react-provider.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import {getAppOrigin} from '@salesforce/pwa-kit-react-sdk/utils/url'
+import {CommerceApiProvider} from '@salesforce/commerce-sdk-react'
+import {UserConfig} from '../types/config'
+
+type WithOptionalCommerceSdkReactProvider = React.ComponentPropsWithoutRef<any>
+
+/**
+ * Higher-order component that conditionally installs the CommerceApiProvider if the config is provided.
+ *
+ * @param WrappedComponent - The component to be optionally wrapped with CommerceApiProvider.
+ * @param config - The configuration object for the CommerceApiProvider.
+ * @returns A component that wraps the given component with CommerceApiProvider if it is not already present in the component tree.
+ */
+export const withOptionalCommerceSdkReactProvider = <P extends object>(
+    WrappedComponent: React.ComponentType<P>,
+    config: UserConfig
+) => {
+    const HOC: React.FC<P> = (props: WithOptionalCommerceSdkReactProvider) => {
+        if (!config.commerceApi || !config.commerceApi?.parameters) {
+            return <WrappedComponent {...(props as P)} />
+        }
```

Comment:
It would be nice if there is a way to "automatically" detect the existence of the `CommerceApiProvider`.

Reply:
I wish the `commerce-sdk-react` throws a "Context not found" error, but currently it throws a weird error 
```
pwa-kit-react-sdk.react-rendering.render ERROR TypeError: Cannot read properties of undefined (reading 'clientConfig')
    at mergeOptions (/Users/kevin.he/dev/pwa-kit/packages/template-typescript-minimal/build/main-server.js:9348:56)
    at useSearchStores (/Users/kevin.he/dev/pwa-kit/packages/template-typescript-minimal/build/main-server.js:7780:81)
```

I'm hesitant to catch that "unofficial" error

Reply:
I'm going to create a ticket to fix it in the sdk

Reply:
As mentioned in our meeting, it is a common pattern to use a hook to see if it throws or not like this:

```
import {useCommerceApi} from '@salesforce/commerce-sdk-react'

const hasCommerceApiProvider = () => {
    const hasProvider = true

    try {
        const api = useCommerceApi() // you can technically use other hooks here, but we might want to stick to using one that has it's own context defined in the lib, and not one relying on react query contexts.
    }
    catch () {
        hasProvider = false
    }

    return hasProvider
} 
```

Then inside this file you can use this locally defined hook to determine if there is already a provider.